website/sampling-consensus.md

```diff
@@ -1,37 +1,79 @@
 
-with `N = 10`, the probability becomes
+Given the demand for inference, ZKML is not scalable enough to offer fast and low cost results. 
 
-`P = (1/3)^10 = 1.6935e-05`,
+##  Optimistic Machine Learning (OPML) 
+
+OPML approaches computations assuming everything is correct unless proven otherwise. Systems operate assuming computations are valid which initially is more efficient because it avoids the cost and delays incurred by immediate verification
+
+In order to verify however, OPML has challenge periods which are specific time windows where participants in the system can challenge the validity of a computation and if a challenge is validated, a correction is made to ensure accuracy. 
+
+These challenge periods lead to delays in finalizing computations because they allow time for challenges to be raised and resolved. The delay can vary in length preventing immediate finality. 
+
+Immediate finality ensures that once a computation is confirmed, it is considered complete and valid across the entire system. Challenge periods delay this finality leading to inconsistent states if challenges alter the initial assumptions. The outcome of a computation might change after assuming correctness which disrupts any subsequent operations. 
+
+
+##  Atoma's Sampling Consensus
+
+Atoma has developed a novel sampling consensus protocol that eliminates the aforementioned issues.   
+
+In order to process requests, the Atoma Network protocol selects a given number of nodes uniformly at random. The selected nodes must execute the given requests, otherwise part of their collateral value is at risk of being slashed. 
+
+The selected nodes are guaranteed to have identical hardware and software specifications and the node sampling mechanism is part of our sampling consensus protocol. 
+
+The Atoma Network sampling consensus protocol satisfies the following: 
+
+1.  Load balancing of the request volume across the Atoma Network. 
+- Requests are only executed across nodes with specific hardware capacities, this node selection homogenizes the balancing of the request load on the Atoma Network.
+
+2.  Nodes reach deterministic agreement on the state of the output generated by executing the given request through the selection mechanism. 
+- All selected nodes will generate the same outputs given the same inputs and the same workload (specified by the incoming request). This is especially important for workloads that rely on floating point arithmetic, the latter being highly non-deterministic.
+- An important example regards AI inference, which depends crucially on floating point arithmetic (specially for LLM inference). By selecting nodes with the same hardware specifications (such as the same GPU card), determinism of floating point arithmetic is guaranteed (with potential mild changes to the kernels which the code runs into). 
```

Comment:
This is great. I think the example is already explicitly mentioned to be AI, so we might replace this with

As such, this is the case for AI

Reply:
Amended